app/models/concerns/pre_sale_prices/line_item_pre_saleable.rb

```diff
@@ -0,0 +1,20 @@
+module PreSalePrices
+  module LineItemPreSaleable
+    extend ActiveSupport::Concern
+
+    included do
+      before_validation :copy_pre_sale_price
```

Comment:
Yeah, I could see that argument. What's more correct, the pre-sale price when the item was first added to basket, or when the product was purchased? 

As you say, the former feels likely more useful for reporting